angular-workspace/projects/ni/nimble-angular/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts

```diff
@@ -13,4 +15,111 @@ describe('Nimble Rich Text Viewer', () => {
             expect(customElements.get('nimble-rich-text-viewer')).not.toBeUndefined();
         });
     });
+
+    describe('with no values in template', () => {
+        @Component({
+            template: `
+                <nimble-rich-text-viewer #viewer></nimble-rich-text-viewer>
+            `
+        })
+        class TestHostComponent {
+            @ViewChild('viewer', { read: NimbleRichTextViewerDirective }) public directive: NimbleRichTextViewerDirective;
+            @ViewChild('viewer', { read: ElementRef }) public elementRef: ElementRef<RichTextViewer>;
+        }
+
+        let fixture: ComponentFixture<TestHostComponent>;
+        let directive: NimbleRichTextViewerDirective;
+        let nativeElement: RichTextViewer;
+
+        beforeEach(() => {
+            TestBed.configureTestingModule({
+                declarations: [TestHostComponent],
+                imports: [NimbleRichTextViewerModule]
+            });
+            fixture = TestBed.createComponent(TestHostComponent);
+            fixture.detectChanges();
+            directive = fixture.componentInstance.directive;
+            nativeElement = fixture.componentInstance.elementRef.nativeElement;
+        });
+
+        it('has expected defaults for markdown', () => {
+            expect(directive.markdown).toBeUndefined();
+            expect(nativeElement.markdown).toBeUndefined();
+        });
+    });
+
+    describe('with template string values', () => {
+        @Component({
+            template: `
+                <nimble-rich-text-viewer #viewer
+                    markdown="**Bold** _Italics_ \n\n1. \n2. \n * \n * \n"
```

Comment:
I'm curious for other reviewers' opinions about whether we should add test coverage for various string values (e.g. values that need escaping) or if we can rely on Angular to do conversions for us.

Reply:
I'm betting the template string values are mostly an edge case and at most have simple initial text. I expect the property binding to be more common and unlikely to cause issues as the value is passed straight through.

I'm okay not being overly robust testing angular's template string behavior. It is whatever angular has chosen and I suspect is documented somewhere. But if folks have issues with it I suspect switching to a binding would be what they try anyway. At the same time not against the tests if wanted :P

Reply:
I have added a test case with the template string literal in the property binding case.